tcms/testcases/views.py

```diff
@@ -603,120 +604,62 @@ def printable(request,  # pylint: disable=missing-permission-required
-    # done by simple_history. Should we remove it ???
-    # Modify the contents
-    fields = ['summary',
-              'case_status',
-              'category',
-              'priority',
-              'notes',
-              'text',
-              'is_automated',
-              'script',
-              'arguments',
-              'extra_link',
-              'requirement']
-
-    for field in fields:
-        if getattr(test_case, field) != tc_form.cleaned_data[field]:
-            setattr(test_case, field, tc_form.cleaned_data[field])
-    try:
-        if test_case.default_tester != tc_form.cleaned_data['default_tester']:
-            test_case.default_tester = tc_form.cleaned_data['default_tester']
-    except ObjectDoesNotExist:
-        pass
-
-    test_case.save()
-
-
-@permission_required('testcases.change_testcase')
-def edit(request, case_id):
-    """Edit case detail"""
-    try:
-        test_case = TestCase.objects.select_related().get(case_id=case_id)
-    except ObjectDoesNotExist:
-        raise Http404
+@method_decorator(permission_required('testcases.change_testcase'), name='dispatch')
+class EditTestCaseView(UpdateView):
+    model = TestCase
+    template_name = 'testcases/mutable.html'
+    form_class = NewCaseForm
 
-    test_plan = plan_from_request_or_none(request)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
 
-    if request.method == "POST":
-        form = NewCaseForm(request.POST)
-        if request.POST.get('product'):
-            form.populate(product_id=request.POST['product'])
-        elif test_plan:
-            form.populate(product_id=test_plan.product_id)
+        if 'notify_form' not in kwargs:
+            context['notify_form'] = CaseNotifyForm(initial={
+                'notify_on_case_update': self.object.emailing.notify_on_case_update,
+                'notify_on_case_delete': self.object.emailing.notify_on_case_delete,
+                'author': self.object.emailing.auto_to_case_author,
+                'default_tester_of_case': self.object.emailing.auto_to_case_tester,
+                'managers_of_runs': self.object.emailing.auto_to_run_manager,
+                'default_testers_of_runs': self.object.emailing.auto_to_run_tester,
+                'assignees_of_case_runs': self.object.emailing.auto_to_case_run_assignee,
+                'cc_list': MultipleEmailField.delimiter.join(
+                    self.object.emailing.get_cc_list()),
+            })
         else:
-            form.populate()
+            context['notify_form'] = kwargs['notify_form']
 
-        n_form = CaseNotifyForm(request.POST)
+        return context
 
-        if form.is_valid() and n_form.is_valid():
-            update_testcase(request, test_case, form)
-            update_case_email_settings(test_case, n_form)
+    def get_form(self, form_class=None):
+        form = super().get_form()
+        test_plan = plan_from_request_or_none(self.request)
```

Comment:
Here I tried to keep functionality backward compatible. The "plan_from_request_or_none" raises 404 exception Im not sure if this is the correct place for that.